test/e2e/kubectl.go

```diff
@@ -566,8 +566,15 @@ var _ = framework.KubeDescribe("Kubectl client", func() {
 			framework.SkipUnlessProviderIs("gke")
 			nsFlag := fmt.Sprintf("--namespace=%v", ns)
 			podJson := readTestFileOrDie(kubectlInPodFilename)
+
+			// Replace the host path to kubectl in json
+			podString := strings.Replace(string(podJson),
+				"$KUBECTL_PATH",
+				framework.TestContext.KubectlPath,
```

Comment:
@ymqytw : then a lot more stuff would break - https://github.com/kubernetes/kubernetes/blob/a7b63e1a959b63f17657ef90414c690fc441d316/test/e2e/framework/util.go#L2001